File: withgoogle/KickStart/2019/Round A/Parcels/AdHocSearch/test/SolutionTests.java
import static org.junit.Assert.*;

import java.io.*;

import org.junit.*;

public class SolutionTests {

    @Test
    public void test0() throws FileNotFoundException {
        InputReader in = new InputReader(new FileInputStream("test_data/0.in"));
        Solution solution = new Solution(in);
        String[] result = solution.solve();
        assertArrayEquals(new String[] {
            "Case #1: 1",
            "Case #2: 0",
            "Case #3: 2"
        }, result);
    }

    @Test
    public void test1() throws FileNotFoundException {
        InputReader in = new InputReader(new FileInputStream("test_data/1.in"));
        Solution solution = new Solution(in);
        String[] result = solution.solve();
        assertArrayEquals(new String[] {
            "Case #1: 4",
            "Case #2: 0",
            "Case #3: 0",
            "Case #4: 2",
            "Case #5: 0",
            "Case #6: 0"
        }, result);
    }

    @Test
    public void test2() throws FileNotFoundException {
        InputReader in = new InputReader(new FileInputStream("test_data/2.in"));
        Solution solution = new Solution(in);
        String[] result = solution.solve();
        assertArrayEquals(new String[] {
            "Case #1: 2",
            "Case #2: 3"
        }, result);
    }
    
    @Test
    public void test3() throws FileNotFoundException {
        InputReader in = new InputReader(new FileInputStream("test_data/3.in"));
        Solution solution = new Solution(in);
        String[] result = solution.solve();
        assertArrayEquals(new String[] {
            "Case #1: 2",
            "Case #2: 4",
            "Case #3: 2",
            "Case #4: 4",
            "Case #5: 2",
            "Case #6: 1"
        }, result);
    }

}

File: withgoogle/KickStart/2019/Round B/Diverse Subarray/SlidingWindowSearch/Solution.Tests/SolutionTests.cs
using System;
using System.IO;
using NUnit.Framework;

public class SolutionTests {

	[Test]
	public void Test0() {
		Solution solution = new Solution(new InputReader(new StreamReader("../../../../test_data/0.in")));
		string[] result = solution.Solve();
		Assert.AreEqual(new string[] {
			"Case #1: 4",
			"Case #2: 6",
			"Case #3: 4",
			"Case #4: 6"
		}, result);
	}

	[Test]
	public void Test1() {
		Solution solution = new Solution(new InputReader(new StreamReader("../../../../test_data/1.in")));
		string[] result = solution.Solve();
		Assert.AreEqual(new string[] {
			"Case #1: 7",
			"Case #2: 5",
			"Case #3: 3"
		}, result);
	}

}
